src/main.c

```diff
@@ -479,38 +481,33 @@ score(char *string)
 static size_t
 min_match_length(char *string)
 {
-	size_t	match_length, match_start, query_position, match_position;
-	int	query_char, query_start;
-
-	query_start = tolower((unsigned char)query[0]);
-
-	for (match_length = 0, match_start = 0; string[match_start] != '\0';
-	    ++match_start) {
-		if (tolower((unsigned char)string[match_start]) ==
-		    query_start) {
-			for (query_position = 1,
-			    match_position = match_start + 1;
-			    query[query_position] != '\0'; ++query_position) {
-				query_char = tolower(
-				    (unsigned char)query[query_position]);
-
-				for (;; ++match_position) {
-					if (string[match_position] == '\0')
-						return match_length;
-
-					if (tolower((unsigned char)string[match_position]) == query_char) {
-						++match_position;
-						break;
-					}
-				}
-			}
-			if (match_length == 0 || match_length > match_position -
-			    match_start + 1)
-				match_length = match_position - match_start + 1;
-		}
-	}
+	char	*start, *end;
+	int	 i;
+	size_t	 length, min;
+
+	if (!(start = strcasechr(string, query[0])))
```

Comment:
In this case, `start` has to be a `char *` to the start of the match but `strncasecmp` returns an `int`. I have updated `strcasechr` to use `strncasecmp`, though in af25c5f.